Wangscape/AlphaCalculator.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <vector>
+#include <functional>
+#include <SFML/Config.hpp>
+class AlphaCalculator
+{
+public:
+    typedef std::vector<float> WeightVector;
+    typedef std::vector<sf::Uint8> AlphaVector;
+    typedef std::function<void(WeightVector&, AlphaVector&)> PixelAlphaCalculator;
+    AlphaCalculator();
+    ~AlphaCalculator();
+    static void calculatePixelAlphaLinear(WeightVector& weights, AlphaVector& alphas);
+    static void calculatePixelAlphaMax(WeightVector& weights, AlphaVector& alphas);
+    static void calculatePixelAlphaFunction(WeightVector& weights, AlphaVector& alphas,
+                                            std::function<float(float)> fn);
+    static PixelAlphaCalculator makeCalculatePixelAlphaPower(float power);
+    static PixelAlphaCalculator bindCalculatePixelAlphaFunction(std::function<float(float)> fn);
```

Comment:
Yes, `AlphaCalculator` has no reason to be a class at the moment. In future I would like to make these calculations composable like `noise::Module`s. Many of these operations (applying lower bound, applying power) should probably be moved into the noise module stack.